docs/syscall_descriptions_syntax.md

```diff
@@ -405,6 +411,155 @@ foo(a ptr[in, array[int8, MY_PATH_MAX]])
 define MY_PATH_MAX	PATH_MAX + 2
 ```
 
+## Conditional fields
```

Comment:
@tarasmadan please review the user documentation with fresh view.
I already know what this is doing, so can't review as a new user would read it.